### PR TITLE
[LANDGRIF-1249]: Wrong behavior in year selector

### PR DESCRIPTION
**Pull request by barbara-chaves**
### General description

This PR fixes the behaviour of the year selector filter on analysis/table

### Testing instructions

- The default last year selected should be the last year with data + 5 years;
- The selectors should allow search values;
- The 'end' year should accept any valid year value bigger then the 'start' year;

## Checklist before merging

- [x] Branch name / PR includes the related Jira ticket Id.
- [x] Tests to check core implementation / bug fix added.
- [x] All checks in Continuous Integration workflow pass.
- [ ] Feature functionally tested by reviewer(s).
- [ ] Code reviewed by reviewer(s).
- [x] Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: client/src/components/forms/select/autocomplete/types.d.ts [hunk 1]
```diff
@@ -16,4 +16,5 @@ export type AutoCompleteSelectProps<T = string> = Omit<
   rowHeight?: number;
   clearable?: boolean;
   onClearSelection?: () => void;
+  onSearch?: (value: string) => void;
```
Comment (andresgnlez) [inline client/src/components/forms/select/autocomplete/types.d.ts:19]: `(value: T) => void;` better, so we can ensure the returned value is of the same type as we set in the options.

Comment (andresgnlez) [reply client/src/components/forms/select/autocomplete/types.d.ts:19]: Anyway, why do you need this `onSearch` callback for? 🤔

File: client/src/containers/analysis-visualization/analysis-filters/years-range/component.tsx [hunk 2]
```diff
@@ -23,37 +27,54 @@ const YearsRange: React.FC = () => {
 
   const { startYear, endYear, yearsGap, setYearsRange } = useYearsRange({
     years,
+    yearsGap: 1,
     // Map mode only makes use of the endYear and will display the Select,
     // not the YearsRangeFilter.
-    validateRange: visualizationMode !== 'map',
+    validateRange: visualizationMode !== 'map' && visualizationMode !== 'table',
```
Comment (andresgnlez) [inline client/src/containers/analysis-visualization/analysis-filters/years-range/component.tsx:33]: Isn't it easier to just validate when `visualizationMode === 'chart'` ?

File: client/src/containers/filters/years-range/component.tsx [hunk 3]
```diff
@@ -52,31 +52,29 @@ export const YearsRangeFilter: React.FC<YearsRangeFilterProps> = ({
...
+  //   onChange?.({ startYear: Number(startYearOption.value), endYear: Number(endYearOption.value) });
+  // }, [
+  //   startYear,
+  //   endYear,
+  //   startYearOptions,
+  //   endYearOption,
+  //   endYearOptions,
+  //   onChange,
+  //   startYearOption,
+  // ]);
```
Comment (andresgnlez) [inline client/src/containers/filters/years-range/component.tsx:77]: Do not leave uncommented code.

File: client/src/containers/analysis-visualization/analysis-filters/years-range/component.tsx (hunk 2 above)
Comment (andresgnlez) [reply client/src/containers/analysis-visualization/analysis-filters/years-range/component.tsx:33]: Anyway, why are we excluding the `table` mode of the validation now? This was not the previous behaviour, wasn't it?

File: client/src/containers/filters/years-range/component.tsx [hunk 4]
```diff
@@ -94,6 +92,10 @@ export const YearsRangeFilter: React.FC<YearsRangeFilterProps> = ({
   // Prevent display when not loaded
   if (!isLoaded) return null;
 
+  const handleChangeEndYear = ({ value }) => {
+    onChange?.({ startYear, endYear: Number(value) });
```
Comment (andresgnlez) [inline client/src/containers/filters/years-range/component.tsx:96]: Aren't we passing the options as `number` ? We shouldn't need to parse the value at this endpoint, it should be a `number` already.

File: client/src/components/forms/select/autocomplete/component.tsx [hunk 5]
```diff
@@ -109,6 +113,9 @@ const AutoCompleteSelect = <T,>({
               })}
             >
               {filteredOptions[index].label}
+              {filteredOptions[index].extraInfo && (
+                <span className="text-sm">{` - ${filteredOptions[index].extraInfo}`}</span>
+              )}
```
Comment (andresgnlez) [inline client/src/components/forms/select/autocomplete/component.tsx:118]: I don't think this solution scales properly, we are assuming here a way to pass extra info to the value but given a certain format (` - {whatever}`), but what if I want to add info with a different format? I just can't.

Anyway, why don't just modify the label of the options?

Comment (barbara-chaves) [reply client/src/components/forms/select/autocomplete/component.tsx:118]: I just used the same property being used before (extraInfo), but I like your solution better. I will change it.

File: client/src/containers/analysis-visualization/analysis-filters/years-range/component.tsx (hunk 2 above)
Comment (barbara-chaves) [reply client/src/containers/analysis-visualization/analysis-filters/years-range/component.tsx:33]: I was testing a solution and add it, then forgot to remove.

File: client/src/containers/filters/years-range/component.tsx [hunk 6]
```diff
@@ -148,23 +130,21 @@ export const YearsRangeFilter: React.FC<YearsRangeFilterProps> = ({
...
-                  onChange={({ value }) => {
-                    onChange?.({ startYear: Number(value), endYear });
-                  }}
+                  onChange={({ value }) => onChange?.({ startYear: value, endYear })}
                   placeholder={placeholderFrom}
                 />
                 <div>To</div>
                 <Select<number>
                   loading={loading}
                   options={endYearOptions.map((option) => ({
```
Comment (andresgnlez) [inline client/src/containers/filters/years-range/component.tsx:139]: Can we move the options to a `useMemo` hook, please? So we avoid mapping them in every render.

**Comment by andresgnlez** [comment]
@barbara-chaves to put the icing on the cake, let's add some tests testing the new behaviors so we ensure we have no regressions.

**Comment by davidsingal** [comment]
This PR should point to staging, could you change it? @barbara-chaves

**Pull request merged**